File: apprefurbished/src/main/java/com/androidcollider/easyfin/managers/rates/rates_info/RatesInfoManager.java
package com.androidcollider.easyfin.managers.rates.rates_info;

import android.content.Context;
import android.view.View;

import com.androidcollider.easyfin.R;
import com.androidcollider.easyfin.managers.ui.toast.ToastManager;
import com.androidcollider.easyfin.repository.Repository;

import rx.Subscriber;

/**
 * @author Ihor Bilous
 */

public class RatesInfoManager {

    private static final long DOUBLE_PRESS_INTERVAL = 1000;
    private long lastPressTime = 0;
    private int count = 0;

    private String info;
    private Context context;
    private Repository repository;
    private ToastManager toastManager;


    RatesInfoManager(Context context, Repository repository, ToastManager toastManager) {
        this.context = context;
        this.repository = repository;
        this.toastManager = toastManager;
        prepareInfo();
    }

    public void setupMultiTapListener(View view, final Context context) {
        view.setOnTouchListener((view1, motionEvent) -> {
            long currentTime = System.currentTimeMillis();
            if (count == 0) lastPressTime = currentTime;

            if (currentTime - lastPressTime < DOUBLE_PRESS_INTERVAL) {
                count++;
                if (count == 7) {
                    toastManager.showClosableToast(context, info, ToastManager.LONG);
                }
            } else count = 0;

            lastPressTime = currentTime;
            return false;
        });
    }

    private void prepareInfo() {
        repository.getRates()
                .subscribe(new Subscriber<double[]>() {

                    @Override
                    public void onCompleted() {

                    }

                    @Override
                    public void onError(Throwable e) {

                    }

                    @Override
                    public void onNext(double[] rates) {
                        String[] currency = context.getResources().getStringArray(R.array.account_currency_array);

                        StringBuilder sb = new StringBuilder();

                        for (int i = 0; i <= rates.length; i++) {
                            sb.append(currency[i]);
                            sb.append(" - ");
                            sb.append(i == 0 ? 1 : rates[i - 1]);
                            sb.append("; ");
                        }

                        info = sb.toString();
                    }
                });
    }
}

File: apprefurbished/src/main/java/com/androidcollider/easyfin/common/managers/rates/rates_info/RatesInfoManager.kt
package com.androidcollider.easyfin.common.managers.rates.rates_info

import android.content.Context
import android.view.MotionEvent
import android.view.View
import com.androidcollider.easyfin.common.managers.resources.ResourcesManager
import com.androidcollider.easyfin.common.managers.ui.toast.ToastManager
import com.androidcollider.easyfin.common.repository.Repository

/**
 * @author Ihor Bilous
 */
class RatesInfoManager internal constructor(
    private val repository: Repository,
    private val toastManager: ToastManager,
    private val resourcesManager: ResourcesManager
) {
    private var lastPressTime: Long = 0
    private var count = 0
    private lateinit var info: String

    init {
        prepareInfo()
    }

    fun setupMultiTapListener(view: View, context: Context) {
        view.setOnTouchListener { view1: View, motionEvent: MotionEvent ->
            val currentTime = System.currentTimeMillis()
            if (count == 0) lastPressTime = currentTime
            if (currentTime - lastPressTime < DOUBLE_PRESS_INTERVAL) {
                count++
                if (count == 7) {
                    toastManager.showClosableToast(context, info, ToastManager.LONG)
                }
            } else count = 0
            lastPressTime = currentTime
            if (motionEvent.action == MotionEvent.ACTION_UP) {
                view1.performClick()
            }
            false
        }
    }

    fun prepareInfo() {
        repository.rates?.subscribe(
            { rates: DoubleArray ->
                val currency =
                    resourcesManager.getStringArray(ResourcesManager.STRING_ACCOUNT_CURRENCY)
                val sb = StringBuilder()
                for (i in 0..rates.size) {
                    sb.append(currency[i])
                    sb.append(" - ")
                    sb.append(if (i == 0) 1 else rates[i - 1])
                    sb.append("; ")
                }
                info = sb.toString()
            }, { obj: Throwable -> obj.printStackTrace() })
    }

    companion object {
        private const val DOUBLE_PRESS_INTERVAL: Long = 1000
    }
}